pkg/controller/deadmanssnitch/deadmanssnitch_controller.go

```diff
@@ -240,6 +240,28 @@ func (r *ReconcileDeadMansSnitch) Reconcile(request reconcile.Request) (reconcil
 			}
 		}
 
+		// Get the snitch again to check status
+		snitches, err = r.dmsclient.FindSnitchesByName(snitchName)
```

Comment:
Is there a reason to re-fetch the snitches? It's done just a few lines above and I don't see anything updating either snitches or snitchName between that line and this one.

Reply:
We refetch the snitch because there is a possibility that a new snitch is created here . https://github.com/openshift/deadmanssnitch-operator/blob/190ec5c31df9b2e96a292ffdc2fa6eff884102c6/pkg/controller/deadmanssnitch/deadmanssnitch_controller.go#L237 .

We want to get the state after the snitch is created to ensure the request has been completed by the API and we aren't just looking at the snitch returned by the create operation, but that it actually exists when pulling as well.